File: 231121/최대 점프 횟수/maximum-number-of-jumps.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int arr[1000];
int dp[1000];
int n, max_val = 0;

int main() {
    cin >> n;
    for(int i = 0; i < n; i++){
        cin >> arr[i];
        dp[i] = -1;
    }
    dp[0] = 0;
    for (int i = 0; i < n; i++){
        for(int j = 0; j < i; j++){
            if (arr[j] + j >= i && dp[j] != -1)
                dp[i] = max(dp[i], dp[j] + 1);
        }
        max_val = max(max_val, dp[i]);
    }
    cout << max_val;
    return 0;
}

File: 231121/최대 점프 횟수/maximum-number-of-jumps.py
n = int(input())
l = list(map(int, input().split()))
dp = [-1 for i in range(n)]
max_val = 0
dp[0] = 0
for i in range(1, n):
    for j in range(i):
        if l[j] + j >= i and dp[j] != -1:
            dp[i] = max(dp[i], dp[j] + 1)
    max_val = max(max_val, dp[i])
print(max_val)